amplfi/utils/result.py

```diff
@@ -112,3 +113,17 @@ def plot_mollview(
         plt.savefig(outpath)
 
         return fig
+
+    def reweight_to_prior(
+        self, target_prior: bilby.core.prior.PriorDict
+    ) -> "AmplfiResult":
+        reweighted = copy(self)
+        keys = list(target_prior.keys())
+        samples_dict = {key: self.posterior[key].values for key in keys}
+        target_probs = target_prior.ln_prob(samples_dict, axis=0)
+        ln_weights = target_probs - self.posterior["log_prior"].values
+        weights = np.exp(ln_weights)
+        reweighted.posterior = bilby.core.result.rejection_sample(
+            reweighted.posterior, weights
+        )
```

Comment:
Looking at the [implementation in bilby](https://git.ligo.org/lscsoft/bilby/-/blob/master/bilby/core/result.py#L282), a single call _will_ lead to a loss in number of samples, which can be drastic at times. You can call this repeatedly until `max_num_rej_sampling_step` times and collect the samples in the way. Something like
```suggestion
        num_orig_samples = len(self.posterior)
        reweighted_samples = np.empty()
        n_step = 1
        while samples.size <= num_orig_samples and n_step < max_num_rej_sampling_step:
           # call rejection sampling
           # append to reweighted samples
        
```